src/Aspire.Hosting.Azure/IKeyVaultResource.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Azure.Security.KeyVault.Secrets;
+
+namespace Aspire.Hosting.Azure;
+
+/// <summary>
+/// Represents a resource that represents an Azure Key Vault.
+/// </summary>
+public interface IKeyVaultResource : IResource, IAzureResource
+{
+    /// <summary>
+    /// Gets the output reference that represents the vault uri for the Azure Key Vault resource.
+    /// </summary>
+    BicepOutputReference VaultUriOutputReference { get; }
+
+    /// <summary>
+    /// Gets the output reference that represents the name of Azure Key Vault resource.
+    /// </summary>
+    BicepOutputReference NameOutputReference { get; }
+
+    /// <summary>
+    /// the output reference that represents the resource id for the Azure Key Vault resource.
+    /// </summary>
+    BicepOutputReference IdOutputReference { get; }
+
+    /// <summary>
+    /// The secret client used to access the Azure Key Vault in run mode.
+    /// </summary>
+    SecretClient? SecretClient { get; set; }
```

Comment:
Do we need this on the interface?

It kind of feels wrong that this `get; set;`

Reply:
Maybe it should be a method? `SetSecretClient` or `InitializeSecretClient`.

Reply:
I just didn't want to introduce another interface.

Reply:
I hate myself for even suggesting - but would IVT work? If we treated it like a public API and didn't make breaking changes to it...

Reply:
I would rather add an interface 😄

Reply:
Made it Func\<...\>, not IVT!